modules/affinity/src/main/java/org/apache/ignite/internal/affinity/ditributed/AffinityManager.java

```diff
@@ -42,45 +41,40 @@
     /** Meta storage service. */
     private MetaStorageManager metaStorageMgr;
 
-    /** Network cluster. */
-    private NetworkCluster networkCluster;
-
-    /** Configuration module. */
-    private ConfigurationManager configurationMgr;
+    /** Configuration manager. */
+    private final ConfigurationManager configurationMgr;
 
     /** Baseline manager. */
-    private BaselineManager baselineMgr;
+    private final BaselineManager baselineMgr;
 
     /** Logger. */
-    private LogWrapper log = new LogWrapper(AffinityManager.class);
+    private final LogWrapper log = new LogWrapper(AffinityManager.class);
 
     /**
      * @param configurationMgr Configuration module.
-     * @param networkCluster Network cluster.
      * @param metaStorageMgr Meta storage service.
      */
     public AffinityManager(
         ConfigurationManager configurationMgr,
-        NetworkCluster networkCluster,
         MetaStorageManager metaStorageMgr,
         BaselineManager baselineMgr
     ) {
         int startRevision =0;
 
         this.configurationMgr = configurationMgr;
-        this.networkCluster = networkCluster;
         this.metaStorageMgr = metaStorageMgr;
         this.baselineMgr = baselineMgr;
 
         String[] metastoragePeerNames = configurationMgr.configurationRegistry()
             .getConfiguration(MetastoreManagerConfiguration.KEY).names().value();
 
-        NetworkMember localMember = networkCluster.localMember();
+        String localMemberName = configurationMgr.configurationRegistry().getConfiguration(LocalConfiguration.KEY)
+            .name().value();
 
         boolean isLocalNodeHasMetasorage = false;
```

Comment:
This is actually a dynamically changed property. It's possible that metastorage topology will change during cluster life cycle. So, eventually, every node should watch MetastoreManagerConfiguration.names in order to detect such dms-topology changes and cancel dms-specific watches if it's no longer a DMS-node or deploy new ones otherwise.

Reply:
I reworked this logic through a listener of the local property (metastorage members).